include/swift/AST/ASTContext.h

```diff
@@ -374,6 +374,10 @@ class ASTContext final {
   llvm::BumpPtrAllocator &
   getAllocator(AllocationArena arena = AllocationArena::Permanent) const;
 
+  /// An optional generic callback function invoked prior to importing a module.
+  mutable std::function<bool(llvm::StringRef ModuleName, bool IsOverlay)>
+      PreModuleImportCallback;
```

Comment:
Nitpick: wrap this with `Optional` to make its optionality explicit.

Reply:
Is that a common pattern in the swift repository? At least in LLVM (and LLDB) I've never seen that. As Ismail pointed out in another comment, `std::function` is empty by default so adding the optional means you have to check both the optional as well as the function before you can call it. That seems needlessly verbose to me.

Reply:
Maybe initializing this inline with `= {}` alleviates your concern?

Reply:
> Maybe initializing this inline with = {} alleviates your concern?

This works too. Thanks!

Reply:
Perfect! I'll go with Adrian's suggestion